File: DI_IOC_Observer/DI_IOC_Observer_JAVA/src/com/kretzer/Triangle.java
package com.kretzer;

import com.sun.javaws.exceptions.InvalidArgumentException;

/**
 * Created by jason for DI_IOC_Observer_JAVA.
 */
public class Triangle extends AbstractPolygon {
    public Triangle(double initSideLength) throws InvalidArgumentException{
        sides = 3;
        setSideLength(initSideLength);
    }
}

File: DI_IOC_Observer/DI_IOC_Observer_CS/DI_IOC_Observer_CS/Triangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DI_IOC_Observer_CS
{
    class Triangle : AbstractPolygon
    {
        public Triangle(double initSideLength) {
            sides = 3;
            setSideLength(initSideLength);
        }
    }
}
